src/sage/schemes/elliptic_curves/hom_composite.py

```diff
@@ -893,3 +896,291 @@ def inseparable_degree(self):
+                    if phi1.domain().j_invariant() == phi2.codomain().j_invariant():
+                        # j-invariant criteria failed
+                        break
+                    last -= 1
+
+            assert first <= last
+
+            # no need to check if only factor ell^1 left
+            if first < last or self._phis[first].degree() % ell**2 == 0:
+                survived = _sieve_torsion_through_isogenies(
+                    self._phis[first:last+1], ell, use_second_curve=_reduce_kernel)
+                if survived.degree() <= 0:
+                    # fully destroyed, non-cyclic
+                    return False
+        return True
+
+
+def _sieve_torsion_through_isogenies(phis, ell, use_second_curve=True):
+    r"""
+    Sieve $\ell$-torsion through isogenies.
+
+    Helper for :meth:`EllipticCurveHom_composite.is_cyclic`.
+    The torsion is represented by the polynomial with
+    x-coordinates as roots (a divisor of the division polynomial
+    of the starting curve).
+
+    INPUT:
+
+    - phis (list of isogenies) -- list of isogenies to propagate the torsion through
+
+    - ell -- an integer defining the torsion $E[\ell]$
+
+    - use_second_curve (optional, default: ``True``): whether to base
+      the remaining torsion representation to the second curve (which is faster
+      because of lower degree).
+
+
+    OUTPUT: a polynomial dividing the division polynomial of the
+    starting curve (or of the second curve if `use_second_curve` is set),
+    representing the torsion subgroup that is not nullified by the isogeny.
+
+    TESTS::
+
+        sage: from sage.schemes.elliptic_curves.hom_composite import _sieve_torsion_through_isogenies
+
+        sage: _sieve_torsion_through_isogenies([], 2)
+        Traceback (most recent call last):
+        ...
+        ValueError: List of isogenies must be nonempty
+
+        sage: E0 = EllipticCurve(GF(37), [4, 0])
+        sage: phi = E0.scalar_multiplication(0)
+        sage: _sieve_torsion_through_isogenies([phi], 2)
+        1
+        sage: phi = E0.scalar_multiplication(2)
+        sage: _sieve_torsion_through_isogenies([phi], 2)
+        1
+        sage: phi = E0.isogeny(E0(0,0)); phi.kernel_polynomial()
+        x
+        sage: _sieve_torsion_through_isogenies([phi], 2) == phi.dual().kernel_polynomial()
+        True
+        sage: _sieve_torsion_through_isogenies([phi], 2, use_second_curve=False).monic()
+        x^2 + 4
+        sage: div = E0.division_polynomial(2); (div // phi.kernel_polynomial()).monic()
+        x^2 + 4
+        sage: _sieve_torsion_through_isogenies([phi], 3).degree()
+        4
+
+    .. SEEALSO::
+
+        - :meth:`sage.schemes.elliptic_curves.hom_composit.EllipticCurveHom_composite.is_cyclic`
+    """
+    if not phis:
+        raise ValueError("List of isogenies must be nonempty")
+    if any(phi.is_zero() or not phi.is_separable() for phi in phis):
+        return phis[0].domain().base_ring()['x'](1)
```

Comment:
```suggestion
        return phis[0].domain().base_ring()['x'].one()
```
(If I understand correctly) this is faster